Modules/Bridge/VtkGlue/test/itkImageToVTKImageFilterTest.cxx

```diff
@@ -20,29 +20,80 @@
 
 #include "itkRandomImageSource.h"
 
+#include "vtkMatrix3x3.h"
+
 int itkImageToVTKImageFilterTest(int, char *[])
 {
-  using ImageType = itk::Image<float, 2 >;
+  const int dim = 3;
+  using ImageType = itk::Image<float, dim >;
   using SourceType = itk::RandomImageSource<ImageType>;
+  using SpacingType = SourceType::SpacingType;
+  using OriginType = SourceType::PointType;
+  using DirectionType = SourceType::DirectionType;
   using ConnectorType = itk::ImageToVTKImageFilter<ImageType>;
 
   ImageType::SizeType size;
-  size.Fill(20);
+  size[0] = 40;
+  size[1] = 10;
+  size[2] = 20;
   SourceType::Pointer source = SourceType::New();
   source->SetSize(size);
 
-  ConnectorType::Pointer connector = ConnectorType::New();
-  connector->SetInput(source->GetOutput());
+  SpacingType spacing;
+  spacing[0] = 0.5;
+  spacing[1] = -2;
+  spacing[2] = 1;
+  OriginType origin;
+  origin[0] = -1.5;
+  origin[1] = 0.222;
+  origin[2] = 0;
+  DirectionType direction;
+  direction.Fill(0.0);
+  direction[0][1] = 1;
+  direction[1][0] = -1;
+  direction[2][2] = 0.7;
+  source->SetSpacing(spacing);
+  source->SetOrigin(origin);
+  source->SetDirection(direction);
 
-  connector->UpdateLargestPossibleRegion();
+  source->Update();
+  auto input = source->GetOutput();
+  input->Print(std::cout);
 
-  connector->GetOutput()->Print(std::cout);
-  connector->GetImporter()->Print(std::cout);
-  connector->GetExporter()->Print(std::cout);
+  ConnectorType::Pointer connector = ConnectorType::New();
+  connector->SetInput(input);
+  connector->UpdateLargestPossibleRegion();
+  connector->Update();
 
-  connector->Print(std::cout);
+  auto output = connector->GetOutput();
+  output->Print(std::cout);
 
-  connector->Update();
+  for (int i = 0; i < dim; ++i)
+  {
+    if (static_cast<int>(input->GetLargestPossibleRegion().GetSize()[i]) != output->GetDimensions()[i])
```

Comment:
It would be safer to cast the VTK dimensions to  `itk::SizeValueType` since it is (at least currently) `unsigned long` which can be larger than `int`, although that doesn't solve the sign issue. Best might be to make sure that `output->GetDimension()[i]` is positive before it is casted, to avoid any possible issue that would be hard to track if a bug was arising from this.